markdown/usage/troubleshooting.md

```diff
@@ -24,6 +28,81 @@ subtitle: How to troubleshoot common mistakes and issues
 - [Using a local version of iGenomes](#using-a-local-version-of-igenomes)
 - [Extra resources and getting help](#extra-resources-and-getting-help)
 
+
+## Troubleshooting basics
+
+These are the recommended steps for troubleshooting a pipeline. 
+
+### Start small
+
+Before using the pipeline with your data, make sure to run a test using: 
+
+```bash
+nextflow run nf-core/<pipeline_name> -profile test,docker
+```
+
+If Docker is not installed, you can replace `docker` with `singularity` or `conda`, see
+the [Getting Started](https://nf-co.re/usage/introduction) tutorial for further information. If a test fails, it might indicate that 
+there is an issue with installation or configuration rather than a pipeline error.
+
+You might also want to check the following:
+
+1. Nextflow is up to date. Use `nextflow self-update` to update a typical installation or `conda update nextflow` for a Bioconda installation.
+2. There is enough disk space, this will avoid running out of space while you are running the pipeline. 
+3. Docker daemon is running (if you are using Docker to manage dependencies).
+
+
+### Categorize the type of error
+
+For this step you try to identify when the error occurs:
+
+1. Before the first process.
+2. During the first process. 
+3. During run.
+4. While generating outputs.
```

Comment:
```suggestion
4. While generating outputs: . Example error:

    ```bash
    [6:16 PM] Error executing process > 'FASTQC (hct116_h3k4me1_IP_R1_T1)' 
    Caused by: 
      Missing output file(s) `*.{zip,html}` expected by process `FASTQC (hct116_h3k4me1_IP_R1_T1)` 
    Command executed: 
      [ ! -f  hct116_h3k4me1_IP_R1_T1.fastq.gz ] && ln -s hct116_h3k4me1_clean.fastq.gz 
    hct116_h3k4me1_IP_R1_T1.fastq.gz 
      fastqc -q -t 6 hct116_h3k4me1_IP_R1_T1.fastq.gz 
    Command exit status: 
      0 
    Command output: 
      (empty) 
    Command error: 
      WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. 
    Memory limited without swap. 
      Failed to process file hct116_h3k4me1_IP_R1_T1.fastq.gz 
      uk.ac.babraham.FastQC.Sequence.SequenceFormatException: Ran out of data in the middle of a 
    fastq entry.  Your file is probably truncated 
       at uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:179) 
       at uk.ac.babraham.FastQC.Sequence.FastQFile.next(FastQFile.java:125) 
       at uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:77) 
       at java.base/java.lang.Thread.run(Thread.java:834)
    ```
```